test/e2e-helm/e2e_helm_metrics_test.go

```diff
@@ -0,0 +1,182 @@
+				}
+				return nil
+			}
+			Eventually(verifyControllerUp, time.Minute, time.Second).Should(Succeed())
+
+			By("Ensuring the created ServiceMonitor for the manager")
+			_, err = tc.Kubectl.Get(
+				true,
+				"ServiceMonitor",
+				fmt.Sprintf("e2e-%s-controller-manager-metrics-monitor", tc.TestSuffix))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Ensuring the created metrics Service for the manager")
+			_, err = tc.Kubectl.Get(
+				true,
+				"Service",
+				fmt.Sprintf("e2e-%s-controller-manager-metrics-service", tc.TestSuffix))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Granting permissions to access the metrics and read the token")
+			_, err = tc.Kubectl.Command(
+				"create",
+				"clusterrolebinding",
+				fmt.Sprintf("metrics-%s", tc.TestSuffix),
+				fmt.Sprintf("--clusterrole=e2e-%s-metrics-reader", tc.TestSuffix),
+				fmt.Sprintf("--serviceaccount=%s:default", tc.Kubectl.Namespace))
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Getting the token")
+			b64Token, err := tc.Kubectl.Get(
+				true,
+				"secrets",
+				"-o=jsonpath={.items[0].data.token}")
+			Expect(err).NotTo(HaveOccurred())
+			token, err := base64.StdEncoding.DecodeString(strings.TrimSpace(b64Token))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(token).NotTo(HaveLen(0))
+
+			By("Pulling docker curl image")
+			pullCurlImage := exec.Command("docker", "pull", curlImage)
+			_, err = tc.Run(pullCurlImage)
+			Expect(err).NotTo(HaveOccurred())
+
+			if IsRunningOnKind() {
+				By("Loading the curl image into the Kind cluster")
+				err = tc.LoadImageToKindClusterWithName(curlImage)
+				Expect(err).Should(Succeed())
+			}
+
+			By("Creating a Pod running the curl image")
+			cmdOpts := []string{
+				"run",
+				"--generator=run-pod/v1",
+				"curl",
+				fmt.Sprintf("--image=%s", curlImage),
+				"--restart=OnFailure",
+				"--", "curl", "-v", "-k", "-H", fmt.Sprintf(`Authorization: Bearer %s`, token),
+				fmt.Sprintf("https://e2e-%v-controller-manager-metrics-service.e2e-%v-system.svc:8443/metrics",
+					tc.TestSuffix, tc.TestSuffix),
+			}
+			_, err = tc.Kubectl.CommandInNamespace(cmdOpts...)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Ensuring curl Pod status")
+			verifyCurlUp := func() error {
+				By("validating the curl pod status")
+				status, err := tc.Kubectl.Get(
+					true,
+					"pods", "curl", "-o", "jsonpath={.status.phase}")
+				Expect(err).NotTo(HaveOccurred())
+				if status != "Completed" && status != "Succeeded" {
+					return fmt.Errorf("curl pod in %s status", status)
+				}
+				return nil
+			}
+			Eventually(verifyCurlUp, 2*time.Minute, time.Second).Should(Succeed())
+
+			By("Checking metrics endpoint serving as expected")
+			getCurlLogs := func() string {
+				logOutput, err := tc.Kubectl.Logs("curl")
+				Expect(err).NotTo(HaveOccurred())
+				return logOutput
+			}
+			Eventually(getCurlLogs, time.Minute, time.Second).Should(ContainSubstring("< HTTP/2 200"))
```

Comment:
Instead of this, I think we should check for the metric produced by the InstrumentedHandler, which would require the metrics test to run while a CR is present.

Reply:
should not it be a follow-up? Are we with the InstrumentedHandler already?

Reply:
IMO: I do not think that we need to remove this check juts add the new after the CR creation.